tools/webassembly.py

```diff
@@ -434,6 +443,37 @@ def get_tables(self):
   def has_name_section(self):
     return self.get_custom_section('name') is not None
 
+  _done_read_imports = False
```

Comment:
can this just be turned into `@cache` too?

Reply:
Its seems like a slightly different use case.. since it doesn't return anything.   But in theory yes I think so..

Reply:
Ah, right, it would be a bit weird. Well, either way is fine; I just noticed that it was asymmetric.